### PR TITLE
Add Baidu AI Cloud

### PR DESCRIPTION
**Pull request by thisiscd**
Baidu AI Cloud focuses on cloud computing, big data, and AI services, providing stable cloud servers, cloud hosts, cloud storage, CDN. The website is ranked 37,988 on [Similarweb](https://pro.similarweb.com/#/digitalsuite/websiteanalysis/overview/website-performance/*/999/1m?webSource=Total&key=cloud.baidu.com). The website provides documentation on how to turn on 2FA.

### CODE REVIEW

File: entries/c/cloud.baidu.com.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "Baidu AI cloud": {
+      "domain": "cloud.baidu.com",
+      "tfa": [
+        "totp",
+        "custom-software"
+      ],
+      "custom-software": [
+        "Baidu AI Cloud APP"
```
Comment (Carlgo11) [inline entries/c/cloud.baidu.com.json]: The documentation page you linked to doesn't mention MFA using Baidu AI Cloud App. Just SMS and TOTP.

File: entries/c/cloud.baidu.com.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      ],
+      "documentation": "https://intl.cloud.baidu.com/doc/IAM/s/2jwvyc50i-en",
+      "categories": [
+        "cloud"
+      ]
```
Comment (Carlgo11) [inline entries/c/cloud.baidu.com.json:17]: Even though the service isn't restricted to mainland China, the login page is mostly in Chinese regardless of the selected language, and there are no English terms of service or privacy policies. Furthermore, there's no mention of GDPR or U.S. equivalents, which would rule out those markets completely. 
I therefore suggest adding a regions element labeling China as the main audience:
```suggestion
      ],
      "regions": [
        "cn"
      ]
```

File: entries/c/cloud.baidu.com.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "Baidu AI cloud": {
+      "domain": "cloud.baidu.com",
```
Comment (Carlgo11) [inline entries/c/cloud.baidu.com.json:3]: The authentication seems to be done at a separate domain.
```suggestion
      "domain": "cloud.baidu.com",
      "additional-domains": [
        "login.bce.baidu.com"
      ],
```

File: entries/c/cloud.baidu.com.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      "tfa": [
+        "totp",
```
Comment (Carlgo11) [inline entries/c/cloud.baidu.com.json:8]:
```suggestion
        "totp",
        "sms",
```

**Review by Carlgo11 — CHANGES_REQUESTED** [review]
Thanks for the pull request @thisiscd!
There are a few things that needs changing before I can merge your contribution.

File: entries/c/cloud.baidu.com.json (hunk 1 above)
Comment (thisiscd) [reply entries/c/cloud.baidu.com.json]: Yes, you are right! I was confused with custom identify authenticator and custom software before. This section [Bind the Virtual MFA](https://intl.cloud.baidu.com/doc/IAM/s/2jwvyc50i-en#bind-the-virtual-mfa) mentions MFA using Baidu AI Cloud App just like normal authenticator.

File: entries/c/cloud.baidu.com.json (hunk 2 above)
Comment (thisiscd) [reply entries/c/cloud.baidu.com.json:17]: Yes, but I found another [domain](https://intl.cloud.baidu.com/) offer English resource to users. How should I deal with it?

Comment (Carlgo11) [reply entries/c/cloud.baidu.com.json:17]: Yeah I found that page as well but my the lack of required legal documents and the login page being 50% Chinese still stands.
Therefore, please include the regions element as suggested.

Comment (thisiscd) [reply entries/c/cloud.baidu.com.json:17]: Thanks for your suggestion!

**Review by Carlgo11 — APPROVED** [review]

**Pull request merged**